src/components/MemoryBar.tsx

```diff
@@ -0,0 +1,38 @@
+import { ProgressBar } from '@/components/ProgressBar';
+import { formatMb } from '@/helpers/numbers';
+
+export function MemoryBar({
+  memoryUsed,
+  memoryAllocated,
+}: {
+  memoryUsed?: number | null | undefined;
+  memoryAllocated?: number | null | undefined;
+}) {
+  const memoryUsedNum = memoryUsed || 0;
+  const mbRemaining = (memoryAllocated || 0) - memoryUsedNum;
+  return (
+    <div className="margin-top-3" data-testid="memory-bar">
+      <p className="font-sans-3xs text-uppercase text-bold">Memory:</p>
+
+      <ProgressBar total={memoryAllocated} fill={memoryUsedNum} />
+
+      <div className="margin-top-1 display-flex flex-justify font-sans-3xs">
+        <div className="margin-right-1">
+          {memoryAllocated && memoryAllocated > 0 && (
+            <>
+              {formatMb(memoryUsedNum)} of {formatMb(memoryAllocated)} allocated
```

Comment:
Should `allocated` be `used` in this case?

Reply:
Allocated is correct here, but it does seem ambiguous. We could make it more explicit by saying "5MB _used_ of 10MB allocated." @beepdotgov What do you think about that?

Reply:
> We could make it more explicit by saying "5MB used of 10MB allocated." 

That phrasing sounds great to me!

(Semi-related: as we shift into apps+spaces, I could use another sync about “memory used” vs “memory allocated”.)